packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -142,7 +142,7 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
       // Fallback for IE
       apiRef.current.unstable_storeRowHeightMeasurement(rowId, ref.current.clientHeight);
     }
-  });
+  }, [apiRef, rowHeight, rowId]);
```

Comment:
Side note: I didn't know that eslint doesn't warn if dependency array is missing in useEffect!
But it makes sense - useCallback is pointless without dependency array, while you might want to run useEffect on every render

Reply:
Actually I left without dependencies on purpose when I did the dynamic row height. The idea was that, since `ResizeObserver` is not available, rerendering the grid would partially mimic its behavior. However, I think this could cause an infinite loop so I'm adding dependencies here.